### PR TITLE
Showtech sonic mgmt framework: Add Management Framework functionality for "show tech-support"

### PR DESCRIPTION
**Pull request by kerry-meyer**
Provide the changes required for supporting the "show-techsupport" command via the SONiC Management Framework front end mechanisms (CLI, REST, and gNOI). The Management Framework functionality implemented by this PR improves on the the capabilities currently provided by the SONiC Click CLI interface via the "show techsupport" command by providing the following additional features:

- User-friendly "help" information describing command syntax details for CLI invocation.
- Ability to invoke the command via REST and gNOI mechanisms.

Unit test results are attached to this PR.

[unit_test_log_0607.txt](https://github.com/Azure/sonic-mgmt-common/files/6613156/unit_test_log_0607.txt)

#### Corequisite PRs:
https://github.com/Azure/sonic-telemetry/pull/78
https://github.com/Azure/sonic-telemetry/pull/79

### CODE REVIEW

**Comment by anshuv-mfst** [comment]
Hi @renukamanavalan - could you please review and merge, thanks

**Review by leeprecy — APPROVED** [review]

File: models/yang/annotations/sonic-showtech-annot.yang [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module sonic-showtech-annot {
```
Comment (lguohan) [inline models/yang/annotations/sonic-showtech-annot.yang:1]: why these yang model not in sonic-yang-model in build image?

**Comment by renukamanavalan** [comment]
Looks good to me.

File: models/yang/annotations/sonic-showtech-annot.yang (hunk 1 above)
Comment (venkatmahalingam) [reply models/yang/annotations/sonic-showtech-annot.yang:1]: We use RPC (to use internal Click command via d-bus infra) for show-tech operation from mgmt-framework and no config-DB update here and hence no YANG model is present in sonic-yang-model in build image.

**Review by venkatmahalingam — APPROVED** [review]

**Comment by renukamanavalan** [comment]
@sujinmkang, Can you please help with the review?
It looks good to me,

**Review by sujinmkang — APPROVED** [review]

**Pull request merged**